nni/retiarii/integration.py

```diff
@@ -0,0 +1,150 @@
+import logging
+import threading
+from typing import *
+
+import json_tricks
+import nni
+from nni.msg_dispatcher_base import MsgDispatcherBase
+from nni.protocol import send, CommandType
+from nni.utils import MetricType
+
+from . import utils
+from .graph import MetricData
+
+
+_logger = logging.getLogger('nni.msg_dispatcher_base')
+
+
+class RetiariiAdvisor(MsgDispatcherBase):
+    """
+    The class is to connect Retiarii components to NNI backend.
+
+    It will function as the main thread when running a Retiarii experiment through NNI.
+    Strategy will be launched as its thread, who will call APIs in execution engine. Execution
+    engine will then find the advisor singleton and send payloads to advisor.
+
+    When metrics are sent back, advisor will first receive the payloads, who will call the callback
+    function (that is a member function in graph listener).
+
+    The conversion advisor provides are minimum. It is only a send/receive module, and execution engine
+    needs to handle all the rest.
+
+    FIXME
+        How does advisor exit when strategy exists?
+
+    Attributes
+    ----------
+    send_trial_callback
```

Comment:
Why `send_trial()` has a callback?

Reply:
IDK. Maybe execution engine wants to use it somehow?

Let's keep that in mind and maybe remove it in future...